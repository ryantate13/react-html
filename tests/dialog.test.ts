import {dialog} from '../src';

describe('dialog', () => {
    it('renders with children', () => {
        expect(dialog({}, 'TEST')).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(dialog({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(dialog()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(dialog({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(dialog({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(dialog({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(dialog({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(dialog({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(dialog({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(dialog({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(dialog({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(dialog({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(dialog({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(dialog({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(dialog({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(dialog({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(dialog({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(dialog({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden="true"', () => {
        expect(dialog({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden="false"', () => {
        expect(dialog({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(dialog({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(dialog({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(dialog({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(dialog({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(dialog({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(dialog({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(dialog({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(dialog({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(dialog({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(dialog({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(dialog({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(dialog({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(dialog({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope="true"', () => {
        expect(dialog({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope="false"', () => {
        expect(dialog({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(dialog({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(dialog({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(dialog({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(dialog({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(dialog({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(dialog({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style="[object Object]"', () => {
        expect(dialog({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex="1"', () => {
        expect(dialog({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(dialog({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(dialog({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(dialog({translate: "no"})).toMatchSnapshot();
    });

    it('renders with open="true"', () => {
        expect(dialog({open: true})).toMatchSnapshot();
    });

    it('renders with open="false"', () => {
        expect(dialog({open: false})).toMatchSnapshot();
    });
});