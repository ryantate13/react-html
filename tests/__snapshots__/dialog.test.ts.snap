// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dialog renders arrays of children 1`] = `
<dialog>
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
</dialog>
`;

exports[`dialog renders with accessKey="a,b,c,d" 1`] = `
<dialog
  accessKey="a,b,c,d"
/>
`;

exports[`dialog renders with autoCapitalize="characters" 1`] = `
<dialog
  autoCapitalize="characters"
/>
`;

exports[`dialog renders with autoCapitalize="none" 1`] = `
<dialog
  autoCapitalize="none"
/>
`;

exports[`dialog renders with autoCapitalize="off" 1`] = `
<dialog
  autoCapitalize="off"
/>
`;

exports[`dialog renders with autoCapitalize="on" 1`] = `
<dialog
  autoCapitalize="on"
/>
`;

exports[`dialog renders with autoCapitalize="sentences" 1`] = `
<dialog
  autoCapitalize="sentences"
/>
`;

exports[`dialog renders with autoCapitalize="words" 1`] = `
<dialog
  autoCapitalize="words"
/>
`;

exports[`dialog renders with children 1`] = `
<dialog>
  TEST
</dialog>
`;

exports[`dialog renders with className="foo bar baz" 1`] = `
<dialog
  className="foo bar baz"
/>
`;

exports[`dialog renders with contentEditable="false" 1`] = `
<dialog
  contentEditable="false"
/>
`;

exports[`dialog renders with contentEditable="true" 1`] = `
<dialog
  contentEditable="true"
/>
`;

exports[`dialog renders with dir="auto" 1`] = `
<dialog
  dir="auto"
/>
`;

exports[`dialog renders with dir="ltr" 1`] = `
<dialog
  dir="ltr"
/>
`;

exports[`dialog renders with dir="rtl" 1`] = `
<dialog
  dir="rtl"
/>
`;

exports[`dialog renders with draggable="false" 1`] = `
<dialog
  draggable="false"
/>
`;

exports[`dialog renders with draggable="true" 1`] = `
<dialog
  draggable="true"
/>
`;

exports[`dialog renders with empty attributes 1`] = `<dialog />`;

exports[`dialog renders with hidden=false 1`] = `
<dialog
  hidden={false}
/>
`;

exports[`dialog renders with hidden=true 1`] = `
<dialog
  hidden={true}
/>
`;

exports[`dialog renders with id="TEST" 1`] = `
<dialog
  id="TEST"
/>
`;

exports[`dialog renders with inputMode="decimal" 1`] = `
<dialog
  inputMode="decimal"
/>
`;

exports[`dialog renders with inputMode="email" 1`] = `
<dialog
  inputMode="email"
/>
`;

exports[`dialog renders with inputMode="none" 1`] = `
<dialog
  inputMode="none"
/>
`;

exports[`dialog renders with inputMode="numeric" 1`] = `
<dialog
  inputMode="numeric"
/>
`;

exports[`dialog renders with inputMode="search" 1`] = `
<dialog
  inputMode="search"
/>
`;

exports[`dialog renders with inputMode="tel" 1`] = `
<dialog
  inputMode="tel"
/>
`;

exports[`dialog renders with inputMode="text" 1`] = `
<dialog
  inputMode="text"
/>
`;

exports[`dialog renders with inputMode="url" 1`] = `
<dialog
  inputMode="url"
/>
`;

exports[`dialog renders with is="foo" 1`] = `
<dialog
  is="foo"
/>
`;

exports[`dialog renders with itemID="https://google.com/" 1`] = `
<dialog
  itemID="https://google.com/"
/>
`;

exports[`dialog renders with itemProp="https://google.com/" 1`] = `
<dialog
  itemProp="https://google.com/"
/>
`;

exports[`dialog renders with itemRef="foo bar" 1`] = `
<dialog
  itemRef="foo bar"
/>
`;

exports[`dialog renders with itemScope=false 1`] = `
<dialog
  itemScope={false}
/>
`;

exports[`dialog renders with itemScope=true 1`] = `
<dialog
  itemScope={true}
/>
`;

exports[`dialog renders with itemType="https://google.com/" 1`] = `
<dialog
  itemType="https://google.com/"
/>
`;

exports[`dialog renders with lang="" 1`] = `
<dialog
  lang=""
/>
`;

exports[`dialog renders with lang="es-419" 1`] = `
<dialog
  lang="es-419"
/>
`;

exports[`dialog renders with open=false 1`] = `
<dialog
  open={false}
/>
`;

exports[`dialog renders with open=true 1`] = `
<dialog
  open={true}
/>
`;

exports[`dialog renders with slot="TEST" 1`] = `
<dialog
  slot="TEST"
/>
`;

exports[`dialog renders with spellCheck="false" 1`] = `
<dialog
  spellCheck="false"
/>
`;

exports[`dialog renders with spellCheck="true" 1`] = `
<dialog
  spellCheck="true"
/>
`;

exports[`dialog renders with style={"color":"#000000"} 1`] = `
<dialog
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`dialog renders with tabIndex=1 1`] = `
<dialog
  tabIndex={1}
/>
`;

exports[`dialog renders with title="TEST" 1`] = `
<dialog
  title="TEST"
/>
`;

exports[`dialog renders with translate="no" 1`] = `
<dialog
  translate="no"
/>
`;

exports[`dialog renders with translate="yes" 1`] = `
<dialog
  translate="yes"
/>
`;

exports[`dialog renders without attributes 1`] = `<dialog />`;
